### PR TITLE
fix(forge install): raise error while installing non-existent repo

### PR DESCRIPTION
**Pull request by SusheendharVijay**
+ minor typo

## Motivation

Fixing #767 + minor typo fix

Left off scoping installs for now and fixed the install first.
lmk of any corrections!

### CODE REVIEW

File: cli/src/cmd/install.rs [hunk 1]
```diff
@@ -79,12 +79,19 @@ pub(crate) fn install(
...
 /// installs the dependency as an ordinary folder instead of a submodule
 fn install_as_folder(dep: &Dependency, libs: &Path) -> eyre::Result<()> {
-    Command::new("git")
+    let result = Command::new("git")
         .args(&["clone", &dep.url, &dep.name])
         .current_dir(&libs)
         .stdout(Stdio::piped())
         .spawn()?
-        .wait()?;
+        .wait();
```
Comment (onbjerg) [inline cli/src/cmd/install.rs:87]: This can be replaced with a `.status()?` instead (same goes for below)

File: cli/src/cmd/install.rs [hunk 2]
```diff
@@ -79,12 +79,19 @@ pub(crate) fn install(
...
+
+    // exiting if the repo doesn't exist
+    if let Ok(status) = result {
+        if status.code() == Some(128) {
+            eyre::bail!("Repo: \"{}\" not found!", &dep.url);
+        }
+    }
```
Comment (onbjerg) [inline cli/src/cmd/install.rs:94]: With the above change this is just `if status.code() == ...`.

An alternative is using `.status()?.exit_ok().map_err(..)?` which would also catch exit code 1 (which is used in some cases by git)

**Review by onbjerg — CHANGES_REQUESTED** [review]
Some minor things :smile:

File: cli/src/cmd/install.rs (hunk 1 above)
Comment (SusheendharVijay) [reply cli/src/cmd/install.rs:87]: oh nice, updating

File: cli/src/cmd/install.rs (hunk 2 above)
Comment (SusheendharVijay) [reply cli/src/cmd/install.rs:94]: Was wondering about other error codes, couldn't find any docs online, making changes!

File: cli/src/cmd/install.rs (hunk 1 above)
Comment (onbjerg) [reply cli/src/cmd/install.rs:87]: Ah, I should have been clearer: both `spawn()?` and `wait()` can be merged into one `.status()?` which returns a `Result<ExitStatus>` (where the `Err` variant is if the process could not be spawned)

**Comment by SusheendharVijay** [comment]
Have made the changes ty @onbjerg, is this what you had in mind? Also, when do we get error code as 1? so I can make a more verbose message

File: cli/src/forge.rs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#![feature(exit_status_error)]
```
Comment (mattsse) [inline cli/src/forge.rs:1]: this makes it `nightly` only

**Review by mattsse — CHANGES_REQUESTED** [review]
instead of checking the status code, can't we scan the stderr output?
this wouldn't require nightly only features

File: cli/src/forge.rs (hunk 3 above)
Comment (gakonst) [reply cli/src/forge.rs:1]: yeah we cant use this

**Comment by gakonst** [comment]
> perhaps we should even start thinking of interacting with git natively again @gakonst ?

depending on libgit made things quite verbose and painful to maintain IMO so would avoid if poss

File: cli/src/cmd/install.rs [hunk 4]
```diff
@@ -105,13 +117,25 @@ fn install_as_folder(dep: &Dependency, libs: &Path) -> eyre::Result<()> {
...
-    Command::new("git")
+    let status = Command::new("git")
         .args(&["submodule", "add", &dep.url, &dep.name])
         .current_dir(&libs)
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
-        .spawn()?
-        .wait()?;
+        .status()?
+        .exit_ok()
```
Comment (gakonst) [inline cli/src/cmd/install.rs:126]: We cannot be using this due to the nightly req https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.exit_ok, could you do the check without the feature
?

**Review by gakonst — CHANGES_REQUESTED** [review]

File: cli/src/cmd/install.rs (hunk 4 above)
Comment (SusheendharVijay) [reply cli/src/cmd/install.rs:126]: Okay sure, making the changes!

**Comment by SusheendharVijay** [comment]
Made the changes, hopefully this is what you had in mind, also added an error if an existing lib was reinstalled, i think that's making some tests fail. lmk about any changes

**Review by onbjerg — APPROVED** [review]
LGTM if this is the direction we want to go in - personally I like using exit codes more, but I understand that this approach gives us more flexibility in what kind of error we can display. All that's left is fixing the tests :smile:

**Comment by gakonst** [comment]
Is the CI error related to this PR?

**Comment by onbjerg** [comment]
Yes, running `forge init --force` in an existing directory fails now if `lib/ds-test` already exists (as it should if you run `forge init && forge init --force`).

Problematic line: https://github.com/gakonst/foundry/blob/0f9fa92e623f079b32e4c4256ac4844020297bf3/cli/src/cmd/init.rs#L124-L125

We should add a check: if we're in a git repo, check if `lib/ds-test` exists, if it doesn't, we install, otherwise we call it with no dependencies (to make sure deps are at least pulled properly)

**Review by gakonst — APPROVED** [review]
nice work!

**Pull request merged**